temporian/io/test/numpy_test.py

```diff
@@ -0,0 +1,81 @@
+import numpy as np
+from absl.testing import absltest
+
+from temporian.implementation.numpy.data.io import event_set
+from temporian.io.numpy import to_numpy
+
+
+class NumpyTest(absltest.TestCase):
+
+    def test_correct(self):
+        evset = event_set(
+            timestamps=['2023-11-08T17:14:38', '2023-11-29T21:44:46'],
+            features={
+                "feature_1": [0.5, 0.6],
+                "my_index": ["red", "blue"],
+            },
+            indexes=["my_index"],
+        )
+
+        result = to_numpy(evset)
+
+        expected_timestamps = np.array(['2023-11-08T17:14:38', '2023-11-29T21:44:46'],
+                                       dtype='datetime64[s]')
+        expected_feature_1 = np.array([0.5, 0.6])
+        expected_my_index = np.array(["red", "blue"])
+
+        # Convert byte strings in x to Unicode strings for a fair comparison
+        result['my_index'] = np.array([item.decode('utf-8') for item in result['my_index']])
+
+        np.testing.assert_array_equal(np.sort(result["feature_1"]), np.sort(expected_feature_1))
+        np.testing.assert_array_equal(np.sort(result["timestamp"]), np.sort(expected_timestamps))
+        np.testing.assert_array_equal(np.sort(result["my_index"]), np.sort(expected_my_index))
```

Comment:
I don't like the `np.sort` here, if the order is not correct we would like this to fail but sorting both arrays before the assert would hide that problem

Reply:
@javiber 
When I create `event_set`, the original input is modified. That is why I am sorting yo make fair comparison. See the below example.


```python
evset = event_set(
            timestamps=['2023-11-08T17:14:38', '2023-11-29T21:44:46'],
            features={
                "feature_1": [0.5, 0.6],
                "my_index": ["red", "blue"],
            },
            indexes=["my_index"],
        )
result = to_numpy(evset)

print(result)


{'my_index': array([b'blue', b'red'], dtype='|S4'), 'feature_1': array([0.6, 0.5]), 
'timestamp': array(['2023-11-29T21:44:46', '2023-11-08T17:14:38'],
 dtype='datetime64[s]')}
```

Please suggest, If I can go ahead like this or make changes.

Reply:
I see, the same thing happens to `to_pandas`. Let's leave it as it is but add a comment saying that we had to sort because to_numpy doesn't follow a particular order (I think it's sorting by index) and I'll open another issue to discuss if we should have a specific order for `to_pandas` and `to_numpy`